core/trino-main/src/test/java/io/trino/sql/planner/iterative/rule/TestPushProjectionIntoTableScan.java

```diff
@@ -185,10 +188,18 @@ public void testPushProjection()
                                                 .build()))));
                     })
                     .matches(project(
-                            newNames.entrySet().stream()
+                            Stream.concat(newNames.entrySet().stream(), constants.entrySet().stream())
                                     .collect(toImmutableMap(
                                             e -> e.getKey().getName(),
-                                            e -> expression(symbolReference(e.getValue())))),
+                                            e -> {
+                                                if (e.getValue() instanceof String) {
```

Comment:
This can use pattern matching to avoid the explicit cast
```java
if (e.getValue() instanceof String value) {
    return expression(symbolReference(value.getValue()));
}

Reply:
Done